backend/create_poll.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"connectrpc.com/connect"
+	zenaov1 "github.com/samouraiworld/zenao/backend/zenao/v1"
+	"go.uber.org/zap"
+)
+
+func (s *ZenaoServer) CreatePoll(ctx context.Context, req *connect.Request[zenaov1.CreatePollRequest]) (*connect.Response[zenaov1.CreatePollResponse], error) {
+	user := s.GetUser(ctx)
+	if user == nil {
+		return nil, errors.New("unauthorized")
+	}
+
+	userID, err := s.EnsureUserExists(ctx, user)
+	if err != nil {
+		return nil, err
+	}
+
+	s.Logger.Info("create-poll", zap.String("question", req.Msg.Question), zap.Strings("options", req.Msg.Options), zap.String("user-id", string(userID)), zap.Bool("user-banned", user.Banned))
+
+	if user.Banned {
+		return nil, errors.New("user is banned")
+	}
+
+	if err := validatePoll(req.Msg.Question, req.Msg.Options /*req.Msg.Kind,*/, req.Msg.Duration); err != nil {
+		return nil, fmt.Errorf("invalid input: %w", err)
+	}
+
+	//XXX: if saving post to db, use a gorm Tx
+	roles, err := s.DB.UserRoles(userID, req.Msg.EventId)
+	if err != nil {
+		return nil, err
+	}
+	if len(roles) == 0 {
+		return nil, errors.New("user is not a member of the event")
+	}
+
+	if err := s.Chain.CreatePoll(userID, req.Msg); err != nil {
+		s.Logger.Error("create-poll", zap.Error(err))
```

Comment:
shouldn't we return the error here ?

Reply:
yes, thanks nice catch: [e544cdf](https://github.com/samouraiworld/zenao/pull/282/commits/e544cdf0ab50b2029493fac9590b9a0d402bdcf9)